tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -344,16 +344,19 @@ static void alt_thread_entry(void)
 		"push {r0, r1};\n\t"
 		"mov r1, r7;\n\t"
 		"mov r0, %0;\n\t"
-		"ldmia r0!, {r4-r7};\n\t"
-		"ldmia r0!, {r4-r7};\n\t"
+		"ldr r4, [r0, #16];\n\t"
 		"mov r8, r4;\n\t"
-		"mov r9, r5;\n\t"
-		"mov r10, r6;\n\t"
-		"mov r11, r7;\n\t"
+		"ldr r4, [r0, #20];\n\t"
+		"mov r9, r4;\n\t"
+		"ldr r4, [r0, #24];\n\t"
+		"mov r10, r4;\n\t"
+		"ldr r4, [r0, #28];\n\t"
+		"mov r11, r4;\n\t"
+		"ldmia r0!, {r4-r7};\n\t"
 		"mov r7, r1;\n\t"
 		"pop {r0, r1};\n\t"
-		:	: "r" (&ztest_thread_callee_saved_regs_container)
-		: "memory"
+		: : "r" (&ztest_thread_callee_saved_regs_container)
```

Comment:
Could you put the output, input, and clobber lists in separate lines, like they appear in GCC manual:

```
: /* no output */
: "r" (&ztest_thread_callee_saved_regs_container)
: "memory", "r4", "r5", "r0", "r1", "r8", "r9", "r10", "r11"
```
? 

While you're at it, could you please add spaces between regs in the first `push` instruction `"memory", "r4", "r5", "r0", "r1", "r8", "r9", "r10", "r11"`?

Reply:
Done